python/tchannel/scheme.py

```diff
@@ -39,23 +39,25 @@ def serialize_body(obj)
 
         def deserialize_body(obj)
     """
-    def type(self):
-        raise NotImplementedError()
+    _type = "raw"
 
-    def serialize_header(self, obj):
-        raise NotImplementedError()
+    def type(self):
+        return self._type
 
     def deserialize_header(self, obj):
-        raise NotImplementedError()
+        return obj
 
-    def serialize_body(self, obj):
-        raise NotImplementedError()
+    def serialize_header(self, obj):
+        return obj
 
     def deserialize_body(self, obj):
-        raise NotImplementedError()
+        return obj
+
+    def serialize_body(self, obj):
+        return obj
 
 
-class JsonArgScheme(ArgScheme):
+class JsonArgScheme(RawArgScheme):
```

Comment:
There isn't any default behavior here, AFAIK that's anticipatory.

Also, creating object hierarchies without a need for type switching is a smell. If the only reason you are making a base class is to share code, you're doing it wrong. You can just as easily compose in like functionality.

Class hierarchies make object responsibilities ambiguous because it tempts you to put anything you want to reuse into the parent. What you should be doing is defining objects with very finite responsibilities, and composing them at runtime.